src/mero_elasticache.erl

```diff
@@ -0,0 +1,158 @@
+%% OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+%%
+%% @doc: This module implements cluster Auto Discovery for memcached
+%% http://docs.aws.amazon.com/AmazonElastiCache/latest/UserGuide/AutoDiscovery.html
+%%
+-module(mero_elasticache).
+
+-author('Miriam Pena <miriam.pena@adroll.com>').
+
+%% API
+-export([get_cluster_config/2]).
+
+%% Just for testing purposes
+-export([request_response/4]).
+
+-define(GET_CLUSTER, <<"config get cluster\n">>).
+
+-type cluster_entry() :: {Host :: string(), Addr :: inet:ip_address(), Port :: pos_integer()}.
+-export_type([cluster_entry/0]).
+
+%%%=============================================================================
+%%% External functions
+%%%=============================================================================
+%% Given an elasticache config Endpoint and port, return parsed list of {host, port} nodes in cluster
+-spec get_cluster_config(string(), integer()) -> {ok, list({string(), integer()})} | {error, Reason :: atom()}.
+get_cluster_config(ConfigHost, ConfigPort) ->
+    LineDefinitions = [banner, version, hosts, crlf, eom],
+    Result = mero_elasticache:request_response(ConfigHost, ConfigPort, ?GET_CLUSTER, LineDefinitions),
+    case parse_cluster_config(proplists:get_value(hosts, Result)) of
+        {error, Reason} ->
+            {error, Reason};
+        {ok, Config} ->
+            {ok, [{Host, Port} || {Host, _IPAddr, Port} <- Config]}
+    end.
+
+%%%=============================================================================
+%%% Internal functions
+%%%=============================================================================
+request_response(Host, Port, Command, Names) ->
+    Opts = [binary, {packet, line}, {active, false}, {recbuf, 5000}],
+    {ok, Socket} = gen_tcp:connect(Host, Port, Opts),
+    ok = gen_tcp:send(Socket, Command),
+    Lines = [{Name, begin
+                        {ok, Line} = gen_tcp:recv(Socket, 0, 10000),
+                        Line
+                    end}
+        || Name <- Names],
+    ok = gen_tcp:close(Socket),
+    Lines.
+
+%% Parse host and version lines to return version and list of {host, port} cluster nodes
+-spec parse_cluster_config(binary()) -> {ok, Config :: [cluster_entry()]} | {error, Reason :: atom()}.
+parse_cluster_config(HostLine) ->
+    %% Strip any newlines
+    Entries = binary:replace(HostLine, <<"\n">>, <<>>),
+
+    %% Break entries on spaces and convert to charlists
+    Entries1 = re:split(Entries, <<" ">>, [{return, list}]),
+
+    %% Parse & validate individual entries
+    parse_cluster_entries(Entries1, []).
+
+parse_cluster_entries([], Accum) ->
+    {ok, lists:reverse(Accum)};
+parse_cluster_entries([H|T], Accum) ->
+    case string:tokens(H, "|") of
+        [Host, IP, Port] ->
+            case inet:parse_ipv4_address(IP) of
+                {ok, IPAddr} ->
+                    case catch erlang:list_to_integer(Port) of
```

Comment:
You should consider using `string:to_integer/1` instead of `catch erlang:list_to_integer/1` here.